compendium/DeclarativeServices/test/TestUpdateConfiguration.cpp

```diff
@@ -0,0 +1,117 @@
+/*=============================================================================
+
+  Library: CppMicroServices
+
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=============================================================================*/
+
+#include "TestFixture.hpp"
+#include "gtest/gtest.h"
+
+#include "TestInterfaces/Interfaces.hpp"
+
+namespace test {
+/**
+   * verify that a service's configuration can be updated after the service is activated 
+   * without deactivating and reactivating the service.
+   */
+TEST_F(tServiceComponent, testUpdateConfig_Modified) //DS_CAI_FTC_1
+{
+  //start bundle
+  cppmicroservices::Bundle testBundle = StartTestBundle("TestBundleDSCAI1");
+
+  // Use DS runtime service to validate the component description
+  scr::dto::ComponentDescriptionDTO compDescDTO =
+    dsRuntimeService->GetComponentDescriptionDTO(testBundle,
+                                                 "sample::TestBundleDSCAI1");
+  EXPECT_EQ(compDescDTO.implementationClass, "sample::TestBundleDSCAI1")
+    << "Implementation class in the returned component description must be "
+       "sample::TestBundleDSCAI1";
+
+  // Use DS runtime service to validate the component state.
+  //It should be in the SATISFIED state because the configuration policy is optional.
+  //and component is delayed
+  auto compConfigs =
+    dsRuntimeService->GetComponentConfigurationDTOs(compDescDTO);
+  EXPECT_EQ(compConfigs.size(), 1ul) << "One default config expected";
+  EXPECT_EQ(compConfigs.at(0).state, scr::dto::ComponentState::SATISFIED);
+
+  //getservice to make component active
+  auto ctxt = framework.GetBundleContext();
```

Comment:
auto context = framework.GetBundleContext(); is defined in the SetUp of tServiceComponent. 
We can just make 'context ' global and use it.

Reply:
I would rather not use globals unless absolutely necessary. framework is defined as a member variable in tServiceComponent. If you just change add a member variable called context and get rid of the auto you can avoid a global variable. Also maybe instead of calling it context you could call it ctx which would be more consistent with the other DS tests.